### PR TITLE
Include default values when validating field conditions

### LINKED ISSUE #8145: Conditions do not work when checked field has a default value that would immediately trigger it
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

**This error only happens when you are creating a new item!**

I have a field with a condition, that checks a toggle field if it is set to `true`. If it is `true`, the field with the condition should not be displayed (Hidden on Detail) and should not be editable (Disable editing value).

I set the toggle field to be `true` by default. If you do this, conditions do not work. If I set it to be `false` by default, conditions work normally.

Conditions do not trigger if the initial state of the **create item UI** has a default value that should trigger it immediately.

### To Reproduce

1. Create 2 toggle fields:
2. Toggle field 1: `toggle1`
3. Toggle field 2: `toggle2`
4. Set `toggle1` to not allow null values, and set the default value to `true`
5. Add the following condition to `toggle2`:

Label: Hidden

Rule:

```json
{
    "toggle1": {
        "_eq": true
    }
}
```

Check `Disable editing value` and `Hidden on Detail`

Now try creating a new item, you will see that the condition on `toggle2` does not work. It should hide `toggle2` initially, until `toggle1` is unchecked and re-hide when checked again.

### What version of Directus are you using?

v9.0.0-rc.93

### What version of Node.js are you using?

14.17.6

### What database are you using?

MariaDB 10.3.31

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

running locally via npm package

### LINKED ISSUE #9552: conditions are not working properly
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

When using conditions in a collection the conditions are not applying to the items.

### To Reproduce

Create a new collection,
Create a `checkbox` item, set default value to false,
Create a string with key `text`, go to conditions add a new rule choosing checkbox, setting value to false, tik `hide on details` and save the settings, got to the collection, text will still be there no matter what the value of the checkbox.

while doing the reverse does work, create checkbox with default value of true, add string item add a rule, checkbox set value to `false` , tik `hide on details`
now when browsing to the collection and one tiks the checkbox the rule does apply,

changing default of checkbox value to `false` will now make de rule not apply anymore
same for if default value is `true`, then rules do not apply,

this makes it very confusing when the rules apply and when not, it seems there is something wrong with Boolean interpretation, and or the rules are not loaded on browsing to the collection item,.. 

so to clarify,

while checkbox default is set to `true` and rules are set to `false` the rules apply.
while checkbox default is set to `true` and rules are set to `true` the rules do not apply.
while checkbox default is set to `false` and rules are set to `false` the rules do not apply.
while checkbox default is set to `false` and rules are set to `true` the rules do not apply

### Errors Shown

no errors are shown.

### What version of Directus are you using?

docker latest

### What version of Node.js are you using?

inside docker container

### What database are you using?

mariadb

### What browser are you using?

chrome, firefox, chromium

### What operating system are you using?

Arch Linux, 5.13.6-zen1-1-zen  x86_64 GNU/Linux

### How are you deploying Directus?

Docker

### PR DESCRIPTION
**Pull request by licitdev**
Closes #8145, resolves #9552.

### CODE REVIEW

File: app/src/components/v-form/v-form.vue [hunk 1]
```diff
@@ -222,6 +222,23 @@ export default defineComponent({
...
+			const defaultValues = computed(() => {
+				return fields.value.reduce(function (acc, field) {
+					if (
+						field.schema?.default_value !== undefined &&
+						// Ignore autoincremented integer PK field
+						!(
+							field.schema.is_primary_key &&
+							field.schema.data_type === 'integer' &&
+							typeof field.schema.default_value === 'string'
+						)
```
Comment (joselcvarela) [inline app/src/components/v-form/v-form.vue:234]: Have you considered functions like `now()` as default values?

Comment (licitdev) [reply app/src/components/v-form/v-form.vue:234]: Nope, I did not. Just tested with a timestamp field with auto modified time (on create & update) and there was no issue. 
Is there any interface that I should take note of?

Comment (joselcvarela) [reply app/src/components/v-form/v-form.vue:234]: It is not specific to the `interface` but to `default value`.
If you choose, for example, type `string` and on `default value` you could put `concat('Direcuts', 'is awesome')`.
Maybe you can split `field.schema.default_value` by `(` and select first position? Not sure if it is the proper way.

Comment (joselcvarela) [reply app/src/components/v-form/v-form.vue:234]: My mistake.
Thank you for notice me and sorry for causing some extra noise. I thought we supported functions as default values but it seems only `now()` is supported which is transformed into `CURRENT_TIMESTAMP`.

That being said, looks great to me 😉

**Review by joselcvarela — APPROVED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**